### PR TITLE
OJ-4141: Add github support

### PR DESCRIPTION
**Pull request by eliwind**

### CODE REVIEW

File: jf_agent/main.py [hunk 1]
```diff
@@ -68,9 +76,11 @@ def main():
 
     jira_config = config.get('jira', {})
     bb_config = config.get('bitbucket', {})
+    git_config = config.get('git', bb_config)
```
Comment (mattklein) [inline jf_agent/main.py]: Looks a little weird. Is it your thinking that we still want to support the `bitbucket` config for backwards compatibility? If so then maybe it'd be a bit clearer to single-line it as:
```
git_config = config.get('git', config.get('bitbucket', {}))
```

File: setup.py [hunk 2]
```diff
@@ -5,13 +5,21 @@
...
     packages=setuptools.find_packages(),
-    install_requires=["jira", "tqdm", "stashy", "pyyaml>=4.2b1", "urllib3==1.24.2", "dateparser"],
+    install_requires=[
+        "jira",
+        "tqdm",
+        "stashy",
+        "pyyaml>=4.2b1",
+        "urllib3==1.24.2",
+        "dateparser",
+        "pygithub @ git+ssh://git@github.com/Jellyfish-AI/PyGithub@ed5ec0ab#egg=pygithub-1.43.2",
```
Comment (mattklein) [inline setup.py]: I don't follow the syntax here -- what's ed5ec0ab? I don't see that hash in our fork of the repo. And what does it mean to have the #egg tag?

File: jf_agent/main.py (hunk 1 above)
Comment (eliwind) [reply jf_agent/main.py]: Yeah, I was trying to be backwards-compatible.  I can one-line it if you think that's clearer.

File: setup.py (hunk 2 above)
Comment (eliwind) [reply setup.py]: https://python-packaging.readthedocs.io/en/latest/dependencies.html#packages-not-on-pypi

Comment (eliwind) [reply setup.py]: That hash seems to be the commit on my branch, not the commit as merged into master.  I'll change this to the master one for clarity.

**Pull request merged**